(maint) fix desc/docs confusion (#141)
puppetlabs/puppet-specifications: language/resource-api/README.md

The `Puppet::ResourceApi.register_type(options)` function takes the following keyword arguments:

* `name`: the name of the resource type.

> * `desc`: a doc string that describes the overall working of the resource type, provides examples, and explains prerequisites and known issues. Can also be written as `docs` for backwards compatibility. This compatibility option will go away in the next major revision.

Reviewer: The term "next major revision" does not work so well since it is a reference to the resource-api next major version as opposed to a revision of the specification. Maybe say "in a future major revision"?

Reviewer: hm.. or is this meant to be a "revision of the spec"  - in which case we should perhaps formalize the versions of the specifications - since the entire repo covers "all the things"...

Author: good point. I'll clarify to mean the Resource API gem.

raises the wider question of whether this is still a good place to keep this document or if it were better to put this spec into the https://github.com/puppetlabs/puppet-resource_api repo and synchronize it there with the different release train branches.

Reviewer: It makes sense for it to be here if there can be alternative implementations of the API, but must then be written to be free of implementation concerns. For example, this could be written as "an implementation may support the older `docs` tag to have the same meaning as `desc`. And add a note that PDK version such and such supports, and is expected to drop in...

This could be pedantic and somewhat silly when there is only one project implementing the specification.
Just my 2c.

Author: There is only one implementation and any compatible other implementation would need to implement this fallback too to be compatible. I'll leave it as it is for now.

> * `desc`: a doc string that describes the overall working of the resource type, provides examples, and explains prerequisites and known issues. Can also be written as `docs` for backwards compatibility. This compatibility option will go away in the next major revision of the library.

Reviewer 2: ```suggestion
* `desc`: a doc string that describes the overall working of the resource type, provides examples, and explains prerequisites and known issues. You can also be write `desc` as `docs` for backwards compatibility. This compatibility option will no longer be available in the next major revision of the library.
```

> * `desc`: a doc string that describes the overall working of the resource type, provides examples, and explains prerequisites and known issues. You can also write `desc` as `docs` for backwards compatibility. This compatibility option will no longer be available in the next major revision of the library.

Reviewer: Will you change this text once the "next major revision of the library" is released?

Author: I think I've been pretty good until now to keep this text and the library in sync. I expect to do so in the foreseeable future, too.
